From aad40f6700622c62b62b6cc29a6ae97645b96698 Mon Sep 17 00:00:00 2001
From: Andrea Corallo <andrea.corallo@arm.com>
Date: Wed, 27 May 2020 17:43:48 +0100
Subject: [PATCH 42/49] arm: Fix unwanted fall-throughs in arm.c

gcc/ChangeLog

2020-05-28  Andrea Corallo  <andrea.corallo@arm.com>

	* config/arm/arm.c (mve_vector_mem_operand): Fix unwanted
	fall-throughs.
---
 gcc/config/arm/arm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gcc/config/arm/arm.c b/gcc/config/arm/arm.c
index 7d8df705bf0..226e57885ec 100644
--- a/gcc/config/arm/arm.c
+++ b/gcc/config/arm/arm.c
@@ -13375,6 +13375,7 @@ mve_vector_mem_operand (machine_mode mode, rtx op, bool strict)
 	    if (val % 4 == 0 && val >= 0 && val <= 1020)
 	      return ((reg_no < LAST_ARM_REGNUM && reg_no != SP_REGNUM)
 		      || (!strict && reg_no >= FIRST_PSEUDO_REGISTER));
+	    return FALSE;
 	  default:
 	    return FALSE;
 	}
-- 
2.29.2

